public/app/store/configureStore.ts

```diff
@@ -38,9 +40,39 @@ export function addRootReducer(reducers: any) {
   Object.assign(rootReducers, ...reducers);
 }
 
+const recursiveCleanState = (state: any, stateSlice: any): boolean => {
```

Comment:
Hm, I would expect cleanup action to simply do 
```ts
{ 
  // state, 
  sliceToClenup: undefined;
}
```

If I have state's slice defined like this:
```ts
interface SomeSlice {
  a: number;
  b: number;
}
```

the recursive way will end up with state being `{a: undefined, b: undefined}` which TS will not pick up, and in the end this can lead to some nasty bugs.

Reply:
Damn, but on the other hand making entire slice undefined makes us define the store type to have slices like `Slice?: SliceType`.

Reply:
Shouldn't the cleanuped state by the `initialState`? Right now when we start the application each slice has a `initialState` so with this cleanup we would introduce 3rd state (with mostly `undefined`) that we need to handle.

Reply:
> Shouldn't the cleanuped state by the `initialState`? Right now when we start the application each slice has a `initialState` so with this cleanup we would introduce 3rd state (with mostly `undefined`) that we need to handle.

No, this will actually be initialState after calling the appReducer with the cleaned state. Just following Dan Abramovs pattern here: https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992

Reply:
> > Shouldn't the cleanuped state by the `initialState`? Right now when we start the application each slice has a `initialState` so with this cleanup we would introduce 3rd state (with mostly `undefined`) that we need to handle.
> 
> No, this will actually become initialState after calling the appReducer with the cleaned state. Just following Dan Abramovs pattern here: https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992

Reply:
> Hm, I would expect cleanup action to simply do
> 
> ```ts
> { 
>   // state, 
>   sliceToClenup: undefined;
> }
> ```
> 
> If I have state's slice defined like this:
> 
> ```ts
> interface SomeSlice {
>   a: number;
>   b: number;
> }
> ```
> 
> the recursive way will end up with state being `{a: undefined, b: undefined}` which TS will not pick up, and in the end this can lead to some nasty bugs.

Maybe the naming is bad here, the stateSlice indicates the prop in state that is assigned with the reducer that you want cleaned. Not some arbitrary prop in state because that will not work.

Reply:
Ah got it so this will trigger the default param for that state which will be the `initialState` or whatever is defined in that reducer. Cool that means 
> Damn, but on the other hand making entire slice undefined makes us define the store type to have slices like Slice?: SliceType.

this also does not apply cause the `undefined` is not actually persisted right?

Reply:
Yes

Reply:
Thanks for explaining that @hugohaggmark, that makes sense now